src/OmniSharp.MSBuild/ProjectLoader.cs

```diff
@@ -127,10 +131,7 @@ private static void SetTargetFrameworkIfNeeded(MSB.Evaluation.Project evaluatedP
                 // do better and potentially allow OmniSharp hosts to select a target framework.
                 targetFramework = targetFrameworks[0];
                 evaluatedProject.SetProperty(PropertyNames.TargetFramework, targetFramework);
-            }
-            else if (!string.IsNullOrWhiteSpace(targetFramework) && targetFrameworks.Length == 0)
-            {
-                targetFrameworks = ImmutableArray.Create(targetFramework);
+                evaluatedProject.ReevaluateIfNecessary();
```

Comment:
Here, we tell the project to re-evaluate itself if it's dirty, which will be the case because we set the `<TargetFramework>` property.